### PR TITLE
[Docs] Reorg of menu/landing page and add FAQs + examples section

### PR DESCRIPTION
**Pull request by stichbury**
[Built docs are here](https://vizro--426.org.readthedocs.build/en/426/)

## Description
I've changed the layout of the menu and index/landing pages very slightly, which you can see here for [Vizro](https://vizro--426.org.readthedocs.build/en/426/) and [Vizro-AI](https://vizro--426.org.readthedocs.build/projects/vizro-ai/en/426/).

* For `vizro-core` I made the menu reflect the diataxis organisation with Tutorials/How-to guides/API ref/Explanation/Examples/ plus contribution and link to Vizro-AI
* For `vizro-ai` I did the same
* I updated the `index.md` for each package so the landing page shows more typical user "journey" links such as "get started" and "why use vizro?"
* Added a section for Examples
    * Added a page about the `examples` folder -> still needs a bit of input from engineering
    * Added a page about third party examples/videos etc -> anyone else want to add anything here @Joseph-Perkins ? 
* Renamed the `development` folder in vizro-core to `contribute` (aligning it with the name it has on the index page and in the menu) and `get-started` to `tutorials` for more diataxis goodness.

Note: Before this merges, I need to create a list of redirects for when we next release, so we can swiftly update breaking links. There shouldn't be too many but I have renamed a couple of folders.

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Comment by maxschulz-COL** [comment]
Hm interesting... I think I liked it more previously, although I am not particularly generally married to the tab header names. Mostly I am was trying, and would still aim to keep us close to the diataxis framework (https://diataxis.fr/), which I think really served us well in general!

> * I renamed the `tutorials` folder to `get-started` (aligning it with the name it has on the index page and in the menu)

I would disagree here, because ideally this will not contain just "easy" things, but will be the section where all tutorials lie. Tutorial being our extended learning journeys to learn a topic E2E.

> * I removed the explanation section from the menu and created a FAQ page inside "get started" instead. Added the "Why Vizro?" page in there as a set of FAQs and also added in questions for compatibility (browser + version)

This would then not fit anymore if we keep the tutorial section as FAQs are not a tutorial. I do admit that the name "Explanation" is maybe not the best, but in general I think we need a section where we can actually explain the background behind things. Maybe we make this section "FAQ", which somewhat stands for explanation, because why would you explain something if it wasn't frequently asked 😂 

> * Added a section for Examples with some dummy text for now

Would this eventually become some sort of gallery?

> * Renamed the `development` folder to `contribute` (aligning it with the name it has on the index page and in the menu)

Sounds good!

**Comment by stichbury** [comment]
Good points @maxschulz-COL and, on reflection, I don't think this is perfect yet either.

However, this is where I do diverge from diataxis a bit. I 🖤  their model of content types but I feel this is more about organisation of content than presentation to the user. That is, a user isn't coming to Vizro docs thinking "I need a how-to guide today" or "where's that explanation section?" but rather, they have problems that they want to solve.

While some of the problems may be "Where's the API ref?" or "Where's a tutorial to learn Vizro" I don't think the organisation/problem overlap works so well for explanation and how-to.

What I think I've messed up in the PR so far is conflate organisation and problem and we've ended up with something that doesn't work either way. I will revert a few of the page moves (e.g. install guide from how-to to get-started, renaming of tutorials folder) but let's consider how to present the user with solutions to their problems while organising the content along diataxis lines.

We have a menu and we have a landing page, and right now I've made the two match each other. How about if we instead use the menu to reflect the main types of content (pretty much as it does now) but then use the landing page to call out specific journeys that solve common user problem "Quickstart tutorial", "FAQs", etc?  

I'll give that a go and let you know when it's ready to look at!

**Comment by maxschulz-COL** [comment]
> I'll give that a go and let you know when it's ready to look at!

Very good points! So maybe it could be an idea to even have the landing page and the content tabs mismatch:
- landing page: guides any user arriving here to solve their problem or question, be it "Why not streamlit", "How do I get started", "Are there cool examples", etc
- Content tabs: more aligned with diataxis, such that we stay clean in how we write certain docs sections, but also such that a user wanting to browse available content of a certain types has an easy time (I often actually find myself wondering: what cool tutorials do they have, or where is an overview of guides about all their features)

**Comment by stichbury** [comment]
OK @maxschulz-COL here's another iteration. See what you think: https://vizro--426.org.readthedocs.build/en/426/

This is definitely getting there! I don't think the landing page is necessarily right yet: the design is a bit off and probably the "typical user problems/interests" aren't all the best choices, but I definitely like the separation between organisation and action.

One big question for me: is "how to install Vizro" a how-to guide or is it a tutorial? I want to have it right at the start of the docs/menu so would favour moving it to be a tutorial, and calling it "get started" but WDYT?

**Comment by maxschulz-COL** [comment]
> OK @maxschulz-COL here's another iteration. See what you think: https://vizro--426.org.readthedocs.build/en/426/
> 
> This is definitely getting there! I don't think the landing page is necessarily right yet: the design is a bit off and probably the "typical user problems/interests" aren't all the best choices, but I definitely like the separation between organisation and action.
> 
> One big question for me: is "how to install Vizro" a how-to guide or is it a tutorial? I want to have it right at the start of the docs/menu so would favour moving it to be a tutorial, and calling it "get started" but WDYT?

Yes I like it a lot - and we can definitely iterate over the "user problems" but this is a good starting point I think!

As for the install guide - for me it is a "How to guide". I think it is already very visible now - from the landing page, and from the search (where the guide is the first hit). For me that would be fine. Just installing a software isn't really a tutorial is it?

**Comment by stichbury** [comment]
> Yes I like it a lot - and we can definitely iterate over the "user problems" but this is a good starting point I think!
Great! Thanks for the feedback. I'll continue to work with this plan then (this task was supposed to add the examples in, but it's changed scope slightly, though at least they are added in draft now). I'll also revise Vizro-AI docs to follow a similar style.

**Comment by stichbury** [comment]
@huong-li-nguyen Please could you take a look at the [page I've added about `examples`](https://vizro--426.org.readthedocs.build/en/426/pages/examples/examples/#vizro-live-demo)? You can't see it in the built version, but in [the markdown](https://github.com/mckinsey/vizro/edit/docs/add-awesome-vizro/vizro-core/docs/pages/examples/examples.md?pr=%2Fmckinsey%2Fvizro%2Fpull%2F426) I've indicated with comments in a couple of places where I could do with a bit of help to get the text complete. 

If you're able, could you drop some notes in for me please? Only when convenient, this isn't urgent. Thank you 🙏

**Comment by stichbury** [comment]
Ready for review -- as noted above, I need some input from engineering, so there's a Vale linter error where I've added some placeholder text. Other tests should pass (if they don't, it's probably not a docs issue).

File: vizro-core/docs/pages/examples/examples.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Vizro features
+The [`examples/features` folder](https://github.com/mckinsey/vizro/tree/main/vizro-core/examples/features) of the `vizro-core` package within [Vizro's GitHub repository](https://github.com/mckinsey/vizro) contains an example that illustrates all Vizro's features. The code is available as a Python script, and there is an alternative `yaml_version` that offers the same example but via YAML configuration.
+
+<!-- TO DO -- I need a bit of help here!
+
+* Anything to add about the YAML config -- I don't think we've documented this much elsewhere in the docs have we?
+* Any differences between the two
+* Preferred way of working?
+-->
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/examples/examples.md:36]: We mention the different methods here: https://vizro.readthedocs.io/en/stable/pages/user-guides/dashboard/#use-dashboard-configuration-options

In general, the pydantic version is always the preferred one. Yaml, json, python dict are just alternative ways to configure it, but in the end they will all be transformed to the pydantic version. I would say for 90% of the users the pydantic version is the way to go, the yaml/json version might be helpful if a machine produces the configuration?

File: vizro-core/docs/pages/examples/examples.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Example in the _dev folder
+
+<!-- TO DO -- I need a bit of help here!
+
+* What is this? Should a user be looking at it? Is the name suitable?
+-->
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/examples/examples.md:43]: No need to mention - it's our developer play example, so can change anytime. There is nothing the user should take a look at here.

File: vizro-core/docs/pages/examples/your-examples.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    * [Use Dash AG Grid within a Vizro app](https://www.youtube.com/watch?v=YvtVcXwQw0E)
+
+## Blog posts
+
+## Examples on GitHub
+<!-- vale off -->
+* [Proof of concept example by viiviandias](https://github.com/viiviandias/poc-vizro/blob/main/brasil_stocks.ipynb)
+* [Amazon sales analysis by Bottleneck44](https://github.com/Bottleneck44/Amazon-Sales-Analysis/blob/main/Amazon-analysis.ipynb)
+* [Opioid interactive map by rmejia41](https://github.com/rmejia41/vizro-opioid-interactive-app/blob/main/src/app.py)
+<!-- vale on -->
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/examples/your-examples.md:30]: Do we need to get their permission to mention them here? I guess not 😅 

A little bit of self-marketing 😄 : https://github.com/huong-li-nguyen/vizro-app-demos

I participate in data viz challenges privately using vizro 😄

**Review by huong-li-nguyen — APPROVED** [review]
LGTM 👍

File: vizro-core/docs/pages/examples/examples.md (hunk 1 above)
Comment (stichbury) [reply vizro-core/docs/pages/examples/examples.md:36]: Thanks! I've added this and linked.

**Comment by stichbury** [comment]
Hi @viiviandias @Bottleneck44!

Thank you for putting your Vizro project on GitHub. We would like to include it in our "best Vizro examples" page within the Vizro documentation. Would you be OK with that? [You can see the draft text here](https://vizro--426.org.readthedocs.build/en/426/pages/examples/your-examples/#examples-on-github). 

Do let us know if you make any other projects or spot them online. We're building up the collection and you're our first third-party mentions.

**Review by maxschulz-COL — APPROVED** [review]

**Pull request merged**